### PR TITLE
Implement activity subscribe method

### LINKED ISSUE #18: Activity subscription
Let's implement `Activity::subscribe` method that creates array of Activity\Visit objects. A person should provide his/hers subscription: later we could check the validity of such action and throw an exception if subscription is not possible.

For now just create "activity visits" according to the activity schedule.

```
$activityVisits = Activity::subscribe(Visitor, Subscription)
```

### PR DESCRIPTION
**Pull request by Lehersun**
Implement method that creates array of Activity\Visit objects

### CODE REVIEW

**Comment by Lehersun** [comment]
How Activity :: subscribe can create an array [Activity => Visits]?
I could only think of this method to return an array [Visit]

File: src/Activity.php [hunk 1]
```diff
@@ -17,4 +22,25 @@ public function getName(): string
...
         return $this->name;
     }
+
+    /**
+     * @param array $schedule
+     */
+    public function setSchedule(array $schedule): void
+    {
+        $this->schedule = $schedule;
+    }
```
Comment (Magomogo) [inline src/Activity.php]: With such a public interface clients should know the internal implementation detail: you should not call `subscribe` until you do `setSchedule`. This is not good.

What do you think if we make `Schedule` object a mandatory dependency of an `Activity`? Requiring it in the constructor: an activity could never exists without its schedule, makes sense.

File: src/Activity.php [hunk 2]
```diff
@@ -17,4 +22,25 @@ public function getName(): string
...
+
+    public function subscribe(Visitor $visitor, Membership $subscription)
+    {
+        $activityVisits = [];
+        $from = $subscription->getStartDay();
+        $to = (clone $from)->add($subscription->getPeriod());
+        foreach ($this->schedule as $day) {
```
Comment (Magomogo) [inline src/Activity.php]: You could call `Schedule::materialize` here.

File: src/Activity/Visit.php [hunk 3]
```diff
@@ -5,17 +5,20 @@
...
 
 class Visit
 {
     private $time;
     private $status;
     private $activity;
+    private $visitor;
 
-    public function __construct(\DateTime $time, Activity $activity)
+    public function __construct(\DateTime $time, string $activity, Visitor $visitor)
```
Comment (Magomogo) [inline src/Activity/Visit.php]: Better depend on an object, it gives better abstraction and public interface for doing things. What is the reason you did that change with activity?

**Review by Magomogo — CHANGES_REQUESTED** [review]

**Comment by Magomogo** [comment]
> How Activity :: subscribe can create an array [Activity => Visits]?
> I could only think of this method to return an array [Visit]

Yes, it should give an array of Activity\Visit objects.

File: src/Activity/Visit.php (hunk 3 above)
Comment (Lehersun) [reply src/Activity/Visit.php]: So I have to create an ActivityVisit class?

Comment (Magomogo) [reply src/Activity/Visit.php]: Just pass it here.

**Comment by Lehersun** [comment]
Ready for check

**Pull request merged**